File: js/stage3/S3_OBJ_01.js
// S3_OBJ_01 - Safe read get(obj, path, fallback)
function get(obj, path, fallback) {
 const keys = path.split(".");
 let current = obj;
 for (const key of keys) {
   if (current == null || !(key in current)) {
     return fallback;
   }
   current = current[key];
 }
 return current;
}
const data = { a: { b: { c: 42 } } };
console.log(get(data, "a.b.c", null)); // 42
console.log(get(data, "a.b.x", "N/A")); // N/A

File: Python/stage3/S3_OBJ_01.py
# S3_OBJ_01 — Safe read
def get(obj, path, fallback=None):
   current = obj
   for key in path.split("."):
       if isinstance(current, dict) and key in current:
           current = current[key]
       else:
           return fallback
   return current
data = {"a": {"b": {"c": 42}}}
print(get(data, "a.b.c"))
print(get(data, "a.b.x", "missing"))